mllam_data_prep/derived_variables.py

```diff
@@ -0,0 +1,459 @@
+
+    if isinstance(toa_radiation, xr.DataArray):
+        # Add attributes
+        toa_radiation.name = "toa_radiation"
+        toa_radiation.attrs["long_name"] = "top-of-the-atmosphere radiation"
+        toa_radiation.attrs["units"] = "W*m**-2"
+
+    return toa_radiation
+
+
+def calculate_hour_of_day(time):
+    """
+    Function for calculating hour of day features with a cyclic encoding
+
+    Parameters
+    ----------
+    time : xr.DataArray or datetime object
+        Time
+
+    Returns
+    -------
+    hour_of_day_cos: xr.DataArray or float
+        cosine of the hour of day
+    hour_of_day_sin: xr.DataArray or float
+        sine of the hour of day
+    """
+    logger.info("Calculating hour of day")
+
+    # Get the hour of the day
+    hour_of_day = time.dt.hour
+
+    # Cyclic encoding of hour of day
+    hour_of_day_cos, hour_of_day_sin = cyclic_encoding(hour_of_day, 24)
+
+    if isinstance(hour_of_day_cos, xr.DataArray):
+        # Add attributes
+        hour_of_day_cos.name = "hour_of_day_cos"
+        hour_of_day_cos.attrs[
+            "long_name"
+        ] = "Cosine component of cyclically encoded hour of day"
+        hour_of_day_cos.attrs["units"] = "1"
+
+    if isinstance(hour_of_day_sin, xr.DataArray):
+        # Add attributes
+        hour_of_day_sin.name = "hour_of_day_sin"
+        hour_of_day_sin.attrs[
+            "long_name"
+        ] = "Sine component of cyclically encoded hour of day"
+        hour_of_day_sin.attrs["units"] = "1"
+
+    return hour_of_day_cos, hour_of_day_sin
+
+
+def calculate_day_of_year(time):
+    """
+    Function for calculating day of year features with a cyclic encoding
+
+    Parameters
+    ----------
+    time : xr.DataArray or datetime object
+        Time
+
+    Returns
+    -------
+    day_of_year_cos: xr.DataArray or float
+        cosine of the day of year
+    day_of_year_sin: xr.DataArray or float
+        sine of the day of year
+    """
+    logger.info("Calculating day of year")
+
+    # Get the day of year
+    day_of_year = time.dt.dayofyear
+
+    # Cyclic encoding of day of year - use 366 to include leap years!
+    day_of_year_cos, day_of_year_sin = cyclic_encoding(day_of_year, 366)
+
+    if isinstance(day_of_year_cos, xr.DataArray):
+        # Add attributes
+        day_of_year_cos.name = "day_of_year_cos"
+        day_of_year_cos.attrs[
+            "long_name"
+        ] = "Cosine component of cyclically encoded day of year"
+        day_of_year_cos.attrs["units"] = "1"
+
+    if isinstance(day_of_year_sin, xr.DataArray):
+        # Add attributes
+        day_of_year_sin.name = "day_of_year_sin"
+        day_of_year_sin.attrs[
+            "long_name"
+        ] = "Sine component of cyclically encoded day of year"
+        day_of_year_sin.attrs["units"] = "1"
+
+    return day_of_year_cos, day_of_year_sin
+
+
+def cyclic_encoding(data_array, da_max):
+    """
+    Cyclic encoding of data
+
+    Parameters
+    ----------
+    da : xr.DataArray
```

Comment:
`da` is common prefix for `xr.DataArray` variables when working with xarray, as is `ds` for `xr.Dataset`s. This is also what you find elsewhere in the codebase :)

Reply:
Yes, but I had named the variable`data_array` in the code but `da` in the docstring, so I was not being consistent.
I have changed the names to `data` instead (will push the new commits soon) since `da` is what is usually used when importing `dask. array` as well :)

Reply:
Yes, that was just what I meant:)